### DISCUSSION TITLE
UI: add a way to override default Relationship search pagination

### DISCUSSION THREAD

**Discussion by AlessioGr** [payloadcms/payload#817] [Feature Requests & Ideas]
I have a relationship to a collection. That collection has a lot of objects.

Now in the UI I can click on the relationship field and search for the object I want to select. However, it only searches the first like 10 objects.
To be able to search all of them there, I first need to scroll down all the way (in the select dropdown of the relationship field) until payload loads in all the objects. Only then can I search all of them.

Sometimes it also only searches the full word and does not include results where the partial word matches. I think that should be changed

**Comment by bigmistqke** [comment]
+1 this is a very annoying bug.

Have been peeking a bit in the source code (not a main contributor). The limit seems to be hard-coded in Relationships, with this lazy-loading pattern, and find-query's default limit is 10 too. Is it even possible to fetch unlimited find-query's right now? If not I believe we should make changes to allow it (`limit: -1`?) and just query all of them at the beginning. Lazyload and autocomplete are not really compatible, I believe.

**Comment by jmikrut** [comment]
Hey all, thanks for the question!

Looks to me that there are three items in this issue:

- relationship field search only showing 10 results

The intended functionality _**should**_ be to search all documents when the search input changes (not just the first 10 returned). Yes, you will only initially load the first 10 from any resulting search, but scrolling down to load more should still be possible right now (it will only load more with the same search term present). The team and I will confirm this, but I believe this is currently the behavior, and the intended behavior.

- Increase the default pagination for any given relationship field (new feature)

This is great. We can certainly build a new feature to increase the pagination that any given relationship field dropdown utilizes. I would say removing the pagination is going to be dangerous, because loading 1,000 docs is going to bog down the UI, but hey, it could be up to the end-user, right? Not opposed... will do some digging here.

- Relationship only searches full word, not including results when partial word matches

Good call. This is legit and will be fixed shortly. Overlaps with issue #806 and will be fixed by solution to #806.

I'm going to convert this issue to a Feature Request Discussion for item 2 above, but if you guys can confirm that my statement in 1 is incorrect, then we need a new issue that says something along the lines of "searching relationship fields only searches the first 10 results". But I believe this is actually working as intended at the moment. Will try and replicate ASAP.

**Comment by bigmistqke** [comment]
> but if you guys can confirm that my statement in 1 is incorrect

Yes, I can confirm. At the moment the autocomplete functionality is not working with lists bigger size then 10. Elements which fall out of the fetched pages are not being found/visualized, only after they are lazy loaded in the drop-down menu.

I might have derailed the conversation a bit... My observations were less about being able to configure the lazy-loading settings of the dropdown, but more that how the lazy-loading is implemented right now, it seems to conflict with auto-complete's functionality.

**Comment by bigmistqke** [comment]
before scrolling to the bottom
![image](https://user-images.githubusercontent.com/10504064/180802238-b6f098b6-2f2e-4ff4-a673-3a566646d379.png)
empty search
![image](https://user-images.githubusercontent.com/10504064/180802303-cd1bd017-fb99-4f96-a0a5-bf0f8a0a3676.png)
after scrolling to the bottom
![image](https://user-images.githubusercontent.com/10504064/180802348-3a89ffb8-113b-49e2-875a-1c910fc7877a.png)
found and visualized the element
![image](https://user-images.githubusercontent.com/10504064/180802391-8d692f82-d4a9-4978-9d14-958bd001f6e0.png)
